src/ORM/DataObject.php

```diff
@@ -1993,16 +1995,21 @@ public function getComponents($componentName, $id = null)
      */
     public function getRelationClass($relationName)
     {
-        // Parse many_many
-        $manyManyComponent = $this->getSchema()->manyManyComponent(static::class, $relationName);
+        // Parse many_many, which can have an array instead of a class name
+        $manyManyComponent = static::getSchema()->manyManyComponent(static::class, $relationName);
```

Comment:
This is just tidy up to match best practices, so the new code below which _also_ follows these best practices